query/node.go

```diff
@@ -1,397 +0,0 @@
-	}
-	return fmt.Sprintf("%s(%s%s)", f.functionName, argumentString, groupString)
-}
-
-// QueryString does the heavy lifting so implementations don't have to.
-func (f functionExpression) QueryString() string {
-	argumentStrings := []string{}
-	for i := range f.arguments {
-		argumentStrings = append(argumentStrings, f.arguments[i].QueryString())
-	}
-	return functionFormatString(argumentStrings, f)
-}
-func (f functionExpression) Name() string {
-	// TODO: deprecate (and remove) this behavior before it becomes permanent
-	if f.functionName == "transform.alias" && len(f.arguments) == 2 {
-		if alias, ok := f.arguments[1].(*stringExpression); ok {
-			return alias.value
-		}
-	}
-	argumentStrings := []string{}
-	for i := range f.arguments {
-		argumentStrings = append(argumentStrings, f.arguments[i].Name())
-	}
-	return functionFormatString(argumentStrings, f)
-}
-
-type annotationExpression struct {
-	content    function.Expression
-	annotation string
-}
-
-func (ae annotationExpression) Evaluate(context function.EvaluationContext) (function.Value, error) {
-	return ae.content.Evaluate(context)
-}
-
-func (ae annotationExpression) QueryString() string {
-	return fmt.Sprintf("%s {%s}", ae.content.QueryString(), ae.annotation)
-}
-func (ae annotationExpression) Name() string {
-	return ae.annotation
-}
-
-// etc nodes
-// ---------
-
-type matcherClause struct {
-	regex *regexp.Regexp
-}
-
-// temporary nodes
-// ---------------
-// These nodes are only present during the parsing step and are not present
-// in the resulting command.
-// There are three types of temporary nodes:
-// * literals (constants in the syntax tree).
-// * lists
-// * evaluation context nodes
-
-type stringLiteral struct {
-	literal string
-}
-
-// list of literals
-type stringLiteralList struct {
-	literals []string
-}
-
-// single tag
-type tagLiteral struct {
-	tag string
-}
-
-// a single operator
-type operatorLiteral struct {
-	operator string
-}
-
-type expressionList struct {
-	expressions []function.Expression
-}
-
-type groupByList struct {
-	list      []string
-	collapses bool
-}
-
-// evaluationContextKey represents a key (from, to, sampleby) for the evaluation context.
-type evaluationContextKey struct {
-	key string
-}
-
-// evaluationContextValue represents a value (date, samplingmode, etc.) for the evaluation context.
-type evaluationContextValue struct {
-	value string
-}
-
-// evaluationContextMap represents a collection of key-value pairs that form the evaluation context.
-type evaluationContextNode struct {
-	Start        int64            // Start of data timerange
-	End          int64            // End of data timerange
-	Resolution   int64            // Resolution of data timerange
-	SampleMethod api.SampleMethod // to use when up/downsampling to match requested resolution
-	assigned     map[string]bool  // a map for knowing which elements of the context have been assigned
-}
-
-// Helper functions for printing
-// =============================
```

Comment:
The "`*Node`" types in general stored intermediate values during parsing. They all implemented `PrintNode()` which gave some diagnostic information, but it wasn't actually being used anywhere meaningful.

Each of the nodes has been replaced by the corresponding type that they wrapped up. For example, instead of storing a `durationExpressionNode` node in the node stack, it stores an `expression.Duration` in the node stack. Instead of storing a `metricFetchExpression` node, it stores an `expression. MetricFetchExpression` (todo: remove `Expression` from the type name).

A handful of nodes were kept around. They're now in `query/parser/node.go` but they don't implement diagnostic printing since non-nodes don't, so it has limited usefulness.